proxy/http/HttpTransact.cc

```diff
@@ -7050,7 +7053,11 @@ HttpTransact::handle_response_keep_alive_headers(State *s, HTTPVersion ver, HTTP
   case KA_CLOSE:
   case KA_DISABLED:
     if (s->client_info.keep_alive != HTTP_NO_KEEPALIVE || (ver == HTTP_1_1)) {
-      heads->value_set(c_hdr_field_str, c_hdr_field_len, "close", 5);
+      if (s->client_info.proxy_connect_hdr) {
```

Comment:
Why do we need to take care of the Proxy-Connection header here?

Reply:
This is to deal with the c_hdr_field_str.  If you look a few lines up, it is either set to Proxy-connection or Connection depending on whether the parent selection path was taken or not.  For H2 I assume we do want to set the proxy-connection (or at least that doesn't trigger the drain logic), so I added a test here to only call the new method in the Connection case.

Reply:
Thanks for the explanation, makes sense.